JS Exception on start

Hello there!
I've tried to launch the new version (1.5.0) through the zip archive and I get this error:

```
A JavaScript error occurred in the main process
Uncaught Exception:
TypeError: Cannot read properties of undefined (reading 'start')
    at Object.<anonymous> (/opt/geforcenow-electron/resources/app.asar/scripts/rpc.js:4:26)
    at Module._compile (node:internal/modules/cjs/loader:1116:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1169:10)
    at Module.load (node:internal/modules/cjs/loader:988:32)
    at Module._load (node:internal/modules/cjs/loader:829:12)
    at Function.c._load (node:electron/js2c/asar_bundle:5:13343)
    at Module.require (node:internal/modules/cjs/loader:1012:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/opt/geforcenow-electron/resources/app.asar/scripts/main.js:4:24)
    at Module._compile (node:internal/modules/cjs/loader:1116:14)
(node:50129) UnhandledPromiseRejectionWarning: Error: Could not connect
    at Socket.onerror (/opt/geforcenow-electron/resources/app.asar/node_modules/discord-rpc/src/transports/ipc.js:32:16)
    at Object.onceWrapper (node:events:510:26)
    at Socket.emit (node:events:390:28)
    at emitErrorNT (node:internal/streams/destroy:157:8)
    at emitErrorCloseNT (node:internal/streams/destroy:122:3)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
```

Compiling from source works so I don't know if I'm missing something.

Still not working:
```
Error launching app
Unable to find Electron app at ***/geforce

Cannot find module '***/geforce/scripts/main.js'. Please verify that the package.json has a valid "main" entry
```

I've checked the 1.4.0 release zip and the `package.json` is missing there too, but it's working so I think the problem is somewhere else. I've tested with discord desktop open and have the same error

> Still not working:
> ```
> Error launching app
> Unable to find Electron app at ***/geforce
> 
> Cannot find module '***/geforce/scripts/main.js'. Please verify that the package.json has a valid "main" entry
> ```
> 
> I've checked the 1.4.0 release zip and the `package.json` is missing there too, but it's working so I think the problem is somewhere else. I've tested with discord desktop open and have the same error

since i want this to be simple for you, go to the github repo and in the master branch take the package.json and put it in your folder replacing it then do npm start

--- a/README.md
+++ b/README.md
@@ -37,10 +37,23 @@ Currently NVIDIA doesn't allow changing the keyboard layout on Linux.
 
 In order to get access to that setting, use one of the UA-spoofing CLI arguments mentioned above.
 
-## Disabling Discord Rich Presence
-Currently it cannot be in settings as in the browser version NVIDIA doesn't have that option.
+## Discord Rich Presence
 
-In order to disable it you must use CLI arguments that are mentioned above.
+Discord RPC should work out-of-the-box in most situations, with no user input required.
+The exceptional cases are documented below.
+
+**Native GFN + Flatpak Discord**
+Run the following commands in a terminal: _([source](https://github.com/flathub/com.discordapp.Discord/wiki/Rich-Precense-(discord-rpc)#unsandboxed-applications))_
+```bash
+mkdir -p ~/.config/user-tmpfiles.d
+echo 'L %t/discord-ipc-0 - - - - app/com.discordapp.Discord/discord-ipc-0' > ~/.config/user-tmpfiles.d/discord-rpc.conf
+systemctl --user enable --now systemd-tmpfiles-setup.service
+```
+
+**Disabling the Discord RPC**
+Currently, the only way to disable it is to use the `--disable-rpc` command-line argument ([documented above](https://github.com/hmlendea/gfn-electron#command-line-arguments))
+
+In order to make this permanent, you can edit the applications desktop file launcher and add that argument to the `Exec` line
 
 # Building from source
 

--- a/README.md
+++ b/README.md
@@ -29,7 +29,7 @@ You can also run `geforcenow` _(or `io.github.hmlendea.geforcenow-electron` if y
 
  - **--spoof-chromeos**: Uses a ChromeOS UserAgent string
  - **--spoof-windows**: Uses a Windows UserAgent string
- - **--disable-rpc**: Disables Discord Rich Presence
+ - **--disable-rpc**: Disables the Discord Rich Presence
 
 ## Changing the keyboard layout
 
@@ -42,15 +42,15 @@ In order to get access to that setting, use one of the UA-spoofing CLI arguments
 Discord RPC should work out-of-the-box in most situations, with no user input required.
 The exceptional cases are documented below.
 
-**Native GFN + Flatpak Discord**
-Run the following commands in a terminal: _([source](https://github.com/flathub/com.discordapp.Discord/wiki/Rich-Precense-(discord-rpc)#unsandboxed-applications))_
+### Native GFN + Flatpak Discord
+Run the following commands in a terminal: *([source](https://github.com/flathub/com.discordapp.Discord/wiki/Rich-Precense-(discord-rpc)#unsandboxed-applications))*
 ```bash
 mkdir -p ~/.config/user-tmpfiles.d
 echo 'L %t/discord-ipc-0 - - - - app/com.discordapp.Discord/discord-ipc-0' > ~/.config/user-tmpfiles.d/discord-rpc.conf
 systemctl --user enable --now systemd-tmpfiles-setup.service
 ```
 
-**Disabling the Discord RPC**
+### Disabling the Discord RPC
 Currently, the only way to disable it is to use the `--disable-rpc` command-line argument ([documented above](https://github.com/hmlendea/gfn-electron#command-line-arguments))
 
 In order to make this permanent, you can edit the applications desktop file launcher and add that argument to the `Exec` line

--- a/package.json
+++ b/package.json
@@ -2,7 +2,7 @@
   "name": "geforcenow-electron",
   "appId": "com.github.hmlendea.${name}",
   "productName": "GeForce NOW",
-  "version": "1.5.0",
+  "version": "1.5.1",
   "description": "A Linux desktop web app for GeForce NOW",
   "main": "scripts/main.js",
   "scripts": {

--- a/scripts/rpc.js
+++ b/scripts/rpc.js
@@ -1,10 +1,7 @@
 const client = require('discord-rich-presence')('963128360219869194');
 
-const pkg = require('../package.json');
-const args = pkg.scripts.start;
-
 function DiscordRPC(title) {
-    if (args.includes('--disable-rpc')) return;
+    if (process.argv.includes("--disable-rpc")) return;
 
     let d;
 

--- a/scripts/rpc.js
+++ b/scripts/rpc.js
@@ -1,7 +1,11 @@
-const client = require('discord-rich-presence')('963128360219869194');
+var client;
 
 function DiscordRPC(title) {
     if (process.argv.includes("--disable-rpc")) return;
+
+    if (!client) {
+        client = require('discord-rich-presence')('963128360219869194');
+    }
 
     let d;
 

